src/utils/cargo_metadata.rs

```diff
@@ -79,11 +79,15 @@ pub(crate) struct Package {
 }
 
 impl Package {
-    pub(crate) fn is_library(&self) -> bool {
-        match self.targets[0].kind.as_slice() {
+    pub(crate) fn library_target(&self) -> Option<&Target> {
+        self.targets.iter().find(|target| match target.kind.as_slice() {
             &[ref kind] if kind == "lib" || kind == "proc-macro" => true,
```

Comment:
```suggestion
            [kind] if kind == "lib" || kind == "proc-macro" => true,
```

I also want to ask - why are there multiple kinds for the target? It looks like there's always one kind in the array, is there a reason `target.kind` isn't a `String` instead of a `Vec<String>`?

Reply:
Dunno, that's what `cargo metadata --format-version=1` gives us.

Reply:
Hmm, strange, `cargo metadata` always gives it as a one-element array.

```
cargo metadata --format-version 1 | jq '.packages | .[].targets | .[].kind | length' | uniq -c
   1007 1
```

Anyway, not a big deal, this code looks fine.